homeassistant/components/homekit/util.py

```diff
@@ -24,17 +24,11 @@ def validate_entity_config(values):
             raise vol.Invalid('The configuration for "{}" must be '
                               ' an dictionary.'.format(entity))
 
-        name = config.get(CONF_NAME)
-        params[CONF_NAME] = \
-            cv.string(name) if name else None
-        manufacturer = config.get(CONF_MANUFACTURER)
-        params[CONF_MANUFACTURER] = \
-            cv.string(manufacturer) if manufacturer else None
-        model = config.get(CONF_MODEL)
-        params[CONF_MODEL] = cv.string(model) if model else None
-        serial_number = config.get(CONF_SERIAL_NUMBER)
-        params[CONF_SERIAL_NUMBER] = \
-            cv.string(serial_number) if serial_number else None
+        for index in (CONF_NAME, CONF_MANUFACTURER,
+                      CONF_MODEL, CONF_SERIAL_NUMBER):
+            value = config.get(index)
+            if value:
+                params[index] = value
```

Comment:
I would prefer to use `key` instead.. `index` is usually used for an int counter through lists, so it might lead to confusion here.

Reply:
`key` is already used in the top level at line 20, any suggestions on what else to use?

Reply:
Missed that. What do you think of changing the top level one to `entity_id` and using key here?
(At least if I haven't missed something again.)